application/src/main/java/org/togetherjava/tjbot/commands/code/CodeMessageAutoDetection.java

```diff
@@ -63,6 +73,29 @@ public void onMessageReceived(MessageReceivedEvent event) {
         codeMessageHandler.addAndHandleCodeMessage(originalMessage, true);
     }
 
+    private boolean isSendByTopHelper(Member member) {
+        final Path configPath = Path.of("config.json");
+
+        try {
+            final Config config = Config.load(configPath);
+            final String[] tagManageRolePatterns = config.getTagManageRolePattern().split("\\|");
+
+            for (final String pattern : tagManageRolePatterns) {
+                for (final Role role : member.getRoles()) {
+                    if (role.getName().matches(pattern)) {
+                        return true;
+                    }
+                }
+            }
+
+        } catch (IOException e) {
+            logger.error("Unable to load the configuration file from path '{}'",
+                    configPath.toAbsolutePath(), e);
+        }
```

Comment:
u don't need to do all this by ur self, u just need to ask for `Config` in this classes constructor. And provide the already created instance at the place of initialization i.e. [Features.java](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/commands/Features.java).

also don't use `tagManageRolePattern`, make ur own entry and use that(also put that in the template file, so next person and fill that accordingly).

also i liked the stream approach, use that if possible

u should give a read to the wiki

Reply:
I understand your request and hope to have implemented it to your satisfaction.